repository/src/test/java/HostedRepositoryTest.java

```diff
@@ -0,0 +1,154 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import java.nio.charset.StandardCharsets;
+
+import java.util.Map;
+import java.util.Properties;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.HandlerList;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.IO;
+
+import org.jbpm.process.workitem.WorkDefinitionImpl;
+import org.jbpm.process.workitem.WorkItemRepository;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.anyOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.*;
+
+public class HostedRepositoryTest {
+
+    private Server server;
+    private ServerConnector connector;
+    private Properties configProperties;
+
+    private static int PORT = 0;
+    private static String DEFAULT_HOST = "localhost";
+
+    @Before
+    public void setUp() {
+        try {
+            // load properties
+            InputStream inProps = this.getClass().getResourceAsStream("testconfig.properties");
+            configProperties = new Properties();
+            configProperties.load(inProps);
+
+            // uncomment this and change builddir and version
+            // to debug locally in IDE
+            //setConfigPropertiesForDebugging();
+
+            server = new Server();
+            connector = new ServerConnector(server);
+            connector.setPort(PORT);
+            server.addConnector(connector);
+
+            HandlerList handlers = new HandlerList();
+            ServletContextHandler context = new ServletContextHandler();
+            ServletHolder defaultServ = new ServletHolder("default",
+                                                          DefaultServlet.class);
+            defaultServ.setInitParameter("resourceBase",
+                                         configProperties.getProperty("builddir") + "/repository-" + configProperties.getProperty("version") + "/");
+            defaultServ.setInitParameter("dirAllowed",
+                                         "true");
+            context.addServlet(defaultServ,
+                               "/");
+            handlers.addHandler(context);
+            server.setHandler(handlers);
+
+            server.start();
+        } catch (Exception e) {
+            fail("Error initializing tests: " + e.getMessage());
+        }
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        try {
+            server.stop();
+        } catch (Exception e) {
+            fail("Unable to stop server: " + e.getMessage());
```

Comment:
I wonder if it makes sense to fail the test at this point. There is nothing anyone could do, except moving on and hoping that someone else won't be trying to open the port again. (But see my other comment above.)